tests/kakuro/TestGameController.java

```diff
@@ -71,60 +72,109 @@ public void testDisconnectDatabase() {
-    model1.board[0][1] = new BoardCell(BoardCell.CellType.FILLED10, 5);
-    model1.board[1][0] = new BoardCell(BoardCell.CellType.FILLED01,-1,4);
-    model1.board[1][1] = new BoardCell(BoardCell.CellType.INPUT,3,-1);
-    model1.board[2][1] = new BoardCell(BoardCell.CellType.INPUT,2,-1);
-    model1.board[1][2] = new BoardCell(BoardCell.CellType.INPUT,1,-1);
-    controller1.model = model1;
-    // Act
-    solved = controller1.solveBoard();
-    // Assert
-    assertEquals(solved, true);
-    // Arrange
-    GameController controller2 = new GameController(10,10, GUI);
-    GameModel model2 = new GameModel(10,10); 
-    model2.initBoard();
-    model2.board[0][1] = new BoardCell(BoardCell.CellType.FILLED10, 5);
-    model2.board[1][0] = new BoardCell(BoardCell.CellType.FILLED01,-1,4);
-    model2.board[1][1] = new BoardCell(BoardCell.CellType.INPUT,3,-1);
-    model2.board[2][1] = new BoardCell(BoardCell.CellType.INPUT,2,-1);
-    model2.board[1][2] = new BoardCell(BoardCell.CellType.INPUT,5,-1);
-    controller2.model = model2;
-    // Act
-    solved = controller2.solveBoard();
-    // Assert
-    assertEquals(solved, false);
-    // Arrange
-    GameController controller3 = new GameController(10,10, GUI);
-    GameModel model3 = new GameModel(10,10); 
-    model3.initBoard();
-    model3.board[0][1] = new BoardCell(BoardCell.CellType.FILLED10, 5);
-    model3.board[1][0] = new BoardCell(BoardCell.CellType.FILLED11,-1,4);
-    model3.board[1][1] = new BoardCell(BoardCell.CellType.INPUT,2,-1);
-    model3.board[2][1] = new BoardCell(BoardCell.CellType.INPUT,3,-1);
-    model3.board[1][2] = new BoardCell(BoardCell.CellType.INPUT,2,-1);
-    controller3.model = model3;            
-    // Act
-    solved = controller3.solveBoard();
-    // Assert
-    assertEquals(solved, false);
+//  @brief Test Valid Board
+    public void testSolveBoard1() {
+        // Arrange
+        int columns = 10;
+        int rows = 10;
+        // Act
+        GameModel model = new GameModel(columns, rows);
+        // Assert
+        assertEquals(model.columns, columns);
+        assertEquals(model.rows, rows);        
+        // Arrange
+        GameController controller1 = new GameController(10,10, GUI);
+        GameModel model1 = new GameModel(10,10); 
+        model1.initBoard();  //initialize board
+        model1.board[0][1] = new BoardCell(BoardCell.CellType.FILLED10, 5);
+        model1.board[1][0] = new BoardCell(BoardCell.CellType.FILLED01,-1,4);
+        model1.board[1][1] = new BoardCell(BoardCell.CellType.INPUT,3,-1);
+        model1.board[2][1] = new BoardCell(BoardCell.CellType.INPUT,2,-1);
+        model1.board[1][2] = new BoardCell(BoardCell.CellType.INPUT,1,-1);
+        controller1.model = model1;
+        // Act
+        solved = controller1.solveBoard();
+        // Assert
+        assertEquals(solved, true);
+    }
+    
+    @Test
+//  @brief Test Invalid Board with one wrong vertical sum
+    public void testSolveBoard2() {       
```

Comment:
@changlmasaki that is better!! but don't use a number but say what is does in the function name it is simplier for the documentation and comprehension like testBoardSolveInvalidBoardWithOneWongVerticalSum or something like that which follows a pattern easy to extend with Valid Invalid Many Good Horizontal etc